### ISSUE TITLE
Let `seed=0` by default in `random.key`

### ISSUE THREAD

**Issue by carlosgmartin** [jax-ml/jax#26554]
Let `seed=0` by default in [`random.key`](https://docs.jax.dev/en/latest/_autosummary/jax.random.key.html). This is by far the most commonly-used value:
- https://github.com/search?type=code&q=jax+random.key%280%29
- https://github.com/search?type=code&q=jax+random.PRNGKey%280%29

I can submit a PR for this.

**Comment by yashk2810** [comment]
OOC, what problem are you trying to solve?

**Comment by carlosgmartin** [comment]
Not really a problem, just a minor convenience. (I see `random.key(0)` everywhere.)

**Comment by yashk2810** [comment]
I would let @froystig reply but I think we want the user to specify what key to use. Just because `key(0)` is used everywhere, doesn't mean it should be the default. Like I am pretty sure there is a good reason, we didn't choose a default :)

**Comment by jakevdp** [comment]
I don't think we should provide a default value for `seed`. Requiring it to be passed by the user is useful as a reminder that calling `key` twice doesn't produce different keys unless you change the seed value. Given the prevalence of implicit state-based RNGs in the numerical computing world, I suspect that might end up being a common mistake if `seed` were not required.

**Comment by froystig** [comment]
I agree with @jakevdp and @yashk2810. Also note that `0` isn't a special seed (as no seed is). I'm going to close this but, as usual, happy to continue discussion.
